Exception when reading ParquetFile row groups in 0.7.1

**What happened**:
Using 0.7.1 an exception occurs when reading some parquet files as groups.
This file only has a single group within.
Exception Snippet:

```
E             File "/Users/nw/dev/src/backup-pipeline/transformer-v2/lambdas/transformer.py", line 65, in transform_parquet_file
E               for index, row_group in enumerate(parquet_file.iter_row_groups()):
E             File "/Users/nw/.virtualenvs/bp-lambdas/lib/python3.8/site-packages/fastparquet/api.py", line 330, in iter_row_groups
E               df = self[i].to_pandas(filters=filters, **kwargs)
E             File "/Users/nw/.virtualenvs/bp-lambdas/lib/python3.8/site-packages/fastparquet/api.py", line 253, in __getitem__
E               new_pf = copy.copy(self)
E             File "/Users/nw/.pyenv/versions/3.8.7/lib/python3.8/copy.py", line 102, in copy
E               return _reconstruct(x, None, *rv)
E             File "/Users/nw/.pyenv/versions/3.8.7/lib/python3.8/copy.py", line 272, in _reconstruct
E               y.__setstate__(state)
E             File "/Users/nw/.virtualenvs/bp-lambdas/lib/python3.8/site-packages/fastparquet/api.py", line 661, in __setstate__
E               self._set_attrs()
E             File "/Users/nw/.virtualenvs/bp-lambdas/lib/python3.8/site-packages/fastparquet/api.py", line 203, in _set_attrs
E               self._dtypes()
E             File "/Users/nw/.virtualenvs/bp-lambdas/lib/python3.8/site-packages/fastparquet/api.py", line 640, in _dtypes
E               if self.pandas_nulls:
E           AttributeError: 'ParquetFile' object has no attribute 'pandas_nulls'
```

**Minimal Complete Verifiable Example**:
This only happens for some parquet files, not all.

The code is essentially:
```python
parquet_file = ParquetFile(file)
for index, row_group in enumerate(parquet_file.iter_row_groups()):
    # do stuff with row_group df here
```

**Anything else we need to know?**:

Version 0.7.0 reads this parquet file successfully.
Breakage looks related to https://github.com/dask/fastparquet/commit/54c4c35a3debb64f3032e599d112299745a5c475

I can't provide the actual parquet file that failed but I can provide the schema:
Schema of a failing parquet file:
```
- schema: REQUIRED
| - Op: BYTE_ARRAY, UTF8, OPTIONAL
| - dms_timestamp: BYTE_ARRAY, UTF8, REQUIRED
| - id: INT64, OPTIONAL
| - u_id: INT64, OPTIONAL
| - lr_id: INT64, OPTIONAL
| - first_name: BYTE_ARRAY, UTF8, OPTIONAL
| - surname: BYTE_ARRAY, UTF8, OPTIONAL
| - dob: INT32, DATE, OPTIONAL
| - mobile: BYTE_ARRAY, UTF8, OPTIONAL
| - created_at: INT64, TIMESTAMP_MILLIS, OPTIONAL
| - updated_at: INT64, TIMESTAMP_MILLIS, OPTIONAL
  - rid: INT64, OPTIONAL
```

More info from parquet-tools:
```
############ file meta data ############
created_by: AWS
num_columns: 12
num_rows: 428
num_row_groups: 1
format_version: 2.0
serialized_size: 1181
```

**Environment**:

- Python version: 3.8
- Operating System: Mac/AWS Lambda
- Install method (conda, pip, source): pip

--- a/fastparquet/api.py
+++ b/fastparquet/api.py
@@ -654,7 +654,8 @@ class ParquetFile(object):
         return dtype
 
     def __getstate__(self):
-        return {"fn": self.fn, "open": self.open, "sep": self.sep, "fmd": self.fmd}
+        return {"fn": self.fn, "open": self.open, "sep": self.sep, "fmd": self.fmd,
+                "pandas_nulls": self.pandas_nulls}
 
     def __setstate__(self, state):
         self.__dict__.update(state)

